sentry_sdk/serializer.py

```diff
@@ -365,113 +356,23 @@ def _serialize_node_impl(
             if not isinstance(obj, string_types):
                 obj = safe_repr(obj)
 
-        # Allow span descriptions to be longer than other strings.
-        #
-        # For database auto-instrumented spans, the description contains
-        # potentially long SQL queries that are most useful when not truncated.
-        # Because arbitrarily large events may be discarded by the server as a
-        # protection mechanism, we dynamically limit the description length
-        # later in _truncate_span_descriptions.
-        if (
-            smart_transaction_trimming
-            and len(path) == 3
-            and path[0] == "spans"
-            and path[-1] == "description"
-        ):
```

Comment:
The real change is changing this if. Because `smart_transaction_trimming` evaluated always to "False" the span description was always truncated with `strip_string`. 

By removing the check for `smart_transaction_trimming` the span description will never be truncated